poppy/instrument.py

```diff
@@ -886,38 +892,43 @@ def _get_weights(self, source=None, nlambda=None, monochromatic=None, verbose=Fa
 
             poppy_core._log.info("Computing wavelength weights using synthetic photometry for %s..." % self.filter)
             band = self._get_synphot_bandpass(self.filter)
+            band_wave = band.waveset
+            band_thru = band(band_wave)
             # choose reasonable min and max wavelengths
-            w_above10 = np.where(band.throughput > 0.10 * band.throughput.max())
+            w_above10 = np.where(band_thru > 0.10 * band_thru.max())
 
-            minwave = band.wave[w_above10].min()
-            maxwave = band.wave[w_above10].max()
-            poppy_core._log.debug("Min, max wavelengths = %f, %f" % (minwave / 1e4, maxwave / 1e4))
+            minwave = band_wave[w_above10].min()
+            maxwave = band_wave[w_above10].max()
+            poppy_core._log.debug("Min, max wavelengths = %f, %f" % (
+                minwave.to_value(units.micron), maxwave.to_value(units.micron)))
 
             wave_bin_edges = np.linspace(minwave, maxwave, nlambda + 1)
             wavesteps = (wave_bin_edges[:-1] + wave_bin_edges[1:]) / 2
             deltawave = wave_bin_edges[1] - wave_bin_edges[0]
+            jwst_area = 25.4 * (units.m * units.m)
```

Comment:
`stsynphot` no longer let you set-and-forget the area like PySynphot does, because I deemed the risk of forgetting and then silently using the wrong value to be not worth the convenience.